### PR TITLE
add kubernetes-base-addons test for traefik

### PR DESCRIPTION
**Pull request by joejulian**
This adds an initial test for kubernetes-base-addons

### CODE REVIEW

**Comment by joejulian** [comment]
Still need to adapt CI.

File: tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/metallb-config.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -euo pipefail
+
+JP='{.status.addresses[?(.type == "InternalIP")].address}'
+NODEIP=$(kubectl get node kind-control-plane -o jsonpath="${JP}")
+SLASH24="${NODEIP%.*}"
```
Comment (jr0d) [inline tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/metallb-config.sh:6]: +1 for bash substring usage

File: tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/exec-test.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    rm ${TMPKUBECONFIG}
+    exit $rc
+}
+
+OUTDIR=$(ARTIFACT:-/tmp)
+SERVER=$(kubectl -n kubeaddons get svc traefik-kubeaddons \
+    -o jsonpath="{.status.loadBalancer.ingress[0].ip}{.status.loadBalancer.ingress[0].hostname}")
+SECRET=$(kubectl -n "${NAMESPACE}" get serviceaccount kuttlaccount -o jsonpath="{ .secrets[0].name }")
+TOKEN=$(kubectl -n "${NAMESPACE}" get secret "${SECRET}" -o go-template="{{.data.token | base64decode }}")
+TMPKUBECONFIG=$(mktemp)
```
Comment (jr0d) [inline tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/exec-test.sh:15]: not even a nit, but his could be done with `kubectl -s ${SERVER} --token=${TOKEN}`

**Review by jr0d — APPROVED** [review]
I've not tried it, but it lgtm.

**Comment by joejulian** [comment]
Converted this to a draft. It's blocked on https://github.com/kudobuilder/kuttl/pull/205

File: tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/exec-test.sh (hunk 2 above)
Comment (joejulian) [reply tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/exec-test.sh:15]: This wasn't working when I tested it. It was using the wrong credentials.

File: tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/10-pod.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  containers:
+    - args:
+      - sh
+      - -c
+      - echo SUCCESS; echo FOO; sleep inf
+      image: busybox
+      imagePullPolicy: Always
+      name: testpod
+  restartPolicy: Never
```
Comment (alejandroEsc) [inline tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/10-pod.yaml]: nit, need a new line here.

File: tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/supportresources.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    traefik.ingress.kubernetes.io/rewrite-target: /
+    ingress.kubernetes.io/protocol: https
+spec:
+  rules:
+    - http:
+        paths:
+          - path: /testpath
+            backend:
+              serviceName: kubernetes
+              servicePort: 443
```
Comment (alejandroEsc) [inline tests/kubernetes-base-addons/addons/traefik/1.7.x/traefik-spdy/supportresources.yaml]: nit new line here also.

**Review by alejandroEsc — APPROVED** [review]
few nits, not important, giving my approval.

**Pull request merged**